File: python/examples/errors/cast.py
# Copyright 2025 © BeeAI a Series of LF Projects, LLC
# SPDX-License-Identifier: Apache-2.0

from beeai_framework.errors import FrameworkError


def get_user() -> None:
    raise FrameworkError("User not found")


try:
    get_user()
except Exception as e:
    err = FrameworkError.ensure(e)
    print(err.get_cause())
    print(err.explain())

File: typescript/examples/errors/cast.ts
/**
 * Copyright 2025 © BeeAI a Series of LF Projects, LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { FrameworkError } from "beeai-framework/errors";

function getUser() {
  throw new Error("User was not found!");
}

try {
  getUser();
} catch (e) {
  const err = FrameworkError.ensure(e);
  console.log(err.dump());
  console.log(err.explain());
}
